programs/survival/potion_master_cleric.sc

```diff
@@ -0,0 +1,169 @@
+		buyB: {id: "minecraft:raw_gold", Count: 1},
+		uses: 0,
+		demand: -20,
+		priceMutiplier: 0.0,
+		specialPrice: 0,
+		rewardExp: 1,
+		xp: 1
+	}',
+	// Levitation
+	'{
+		sell: {id: "minecraft:%s", Count:1, tag:{
+			CustomPotionEffects:[{Duration:300, Id: 25, Amplifier: 3}], 
+			CustomPotionColor: 16777215, display: {Name:\'{ "text":"Potion of Levitation"}\'}}
+		},
+		buy: {id: "minecraft:emerald", Count: %d},
+		buyB: {id: "minecraft:phantom_membrane", Count: 1},
+		uses: 0,
+		demand: -20,
+		priceMutiplier: 0.0,
+		specialPrice: 0,
+		rewardExp: 1,
+		xp: 1
+	}',
+	// Wither
+	'{
+		sell: {id: "minecraft:%s", Count:1, tag:{
+			CustomPotionEffects:[{Duration:170, Id: 20, Amplifier: 3}], 
+			CustomPotionColor: 0, display: {Name:\'{ "text":"Potion of Wither"}\'}}
+		},
+		buy: {id: "minecraft:emerald", Count: %d},
+		buyB: {id: "minecraft:wither_rose", Count: 1},
+		uses: 0,
+		demand: -20,
+		priceMutiplier: 0.0,
+		specialPrice: 0,
+		rewardExp: 1,
+		xp: 1
+	}',
+	// Mining Fatigue
+	'{
+		sell: {id: "minecraft:%s", Count:1, tag:{
+			CustomPotionEffects:[{Duration:2800, Id: 4, Amplifier: 0}], 
+			CustomPotionColor: 5592405, display: {Name:\'{ "text":"Potion of Mining Fatigue"}\'}}
+		},
+		buy: {id: "minecraft:emerald", Count: %d},
+		buyB: {id: "minecraft:soul_sand", Count: 1},
+		uses: 0,
+		demand: -20,
+		priceMutiplier: 0.0,
+		specialPrice: 0,
+		rewardExp: 1,
+		xp: 1
+	}',
+	// Blindness
+	'{
+		sell: {id: "minecraft:%s", Count:1, tag:{
+			CustomPotionEffects:[{Duration:330, Id: 15, Amplifier: 0}], 
+			CustomPotionColor: 70000, display: {Name:\'{ "text":"Potion of Blindness"}\'}}
+		},
+		buy: {id: "minecraft:emerald", Count: %d},
+		buyB: {id: "minecraft:ender_eye", Count: 1},
+		uses: 0,
+		demand: -20,
+		priceMutiplier: 0.0,
+		specialPrice: 0,
+		rewardExp: 1,
+		xp: 1
+	}'
+	
+];
+
+_add_trades(cleric, p, nbt) -> (
+	// add 0-max number of potion trades
+	loop(rand(1 + global_max_potion_trades),
+
+		trade = nbt(str(
+			// select a random trade from the list above
+			global_potion_trades:rand(length(global_potion_trades)),
+			// Randomize the potion type 
+			global_potion_types:rand(length(global_potion_types)),
+			// randomize the cost of each trade
+			rand((global_potion_emerald_cost:1) - global_potion_emerald_cost:0) + global_potion_emerald_cost:0
+		));
+
+		// add the new potion trade to the end of the clerics trades
+    	put(nbt, 'Offers.Recipes', trade, -1 );
+	);
+	
+	modify(cleric, 'nbt_merge', nbt);
+	// Dont add any more potions to a potion master
+	modify(cleric, 'tag', 'IsPotionMaster'); 
+
+);
+
+__on_player_interacts_with_entity(player, entity, hand) -> (
+	if(entity~'type'=='villager',
+		nbt = entity~'nbt';
+		// If Villager is a cleric has a high enough level and is not already a potion master
+		if(nbt:'VillagerData':'profession'=='minecraft:cleric' && 
+			nbt:'VillagerData':'level' >= global_potion_master_level &&
+			parse_nbt(nbt:'Tags')~'IsPotionMaster' == null
```

Comment:
Comon practice is to give your tags a namespace specific to the app that ensures that no other app or datapck will colide with yours. For example, in this case I'd use something like `PotionMaster:IsPotionMaster` or if you want it less verbose `PMC:IsPotionMaster`.